lib/spack/spack/environment.py

```diff
@@ -125,7 +125,7 @@ def activate(
             cmds += 'setenv SPACK_OLD_PROMPT "${prompt}";\n'
             cmds += 'set prompt="%s ${prompt}";\n' % prompt
     else:
-        if 'color' in os.environ['TERM'] and prompt:
+        if prompt and 'color' in os.environ.get('TERM', ''):
```

Comment:
As long as we're changing this line, can we move this conditional inside the lower `if prompt` block? That'll keep things a bit neater.

Reply:
That makes sense, I've updated it in the latest commit. 
Thanks